sql/catalyst/src/main/java/org/apache/spark/sql/sources/v2/SupportsDelete.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.sources.v2;
+
+import org.apache.spark.sql.sources.Filter;
+
+public interface SupportsDelete {
```

Comment:
This should include the class comments from the original interface as well (updated for recent changes):

```
/**
 * A mix-in interface for {@link DataSourceV2} delete support. Data sources can implement this
 * interface to provide the ability to delete data from tables that matches filter expressions.
 * <p>
 * Data sources must implement this interface to support logical operations that combine writing
 * data with deleting data, like overwriting partitions.
 */
```

Reply:
Thanks @rdblue , updated.